### ISSUE TITLE
what am i doing wrong? with updater.py not installing ultima scraper?

### ISSUE THREAD

**Issue by wooki0** [DIGITALCRIMINALS/UltimaScraper#890]
PS H:\OnlyFans-master> (Invoke-WebRequest -Uri https://install.python-poetry.org -UseBasicParsing).Content | py -
Retrieving Poetry metadata

The latest version (1.4.0) is already installed.
PS H:\OnlyFans-master> python updater.py                                                                                Creating virtualenv ultima-scraper-ydjPcDtY-py3.10 in C:\Users\xxxxx\AppData\Local\pypoetry\Cache\virtualenvs
Updating dependencies
Resolving dependencies... (3.0s)

Writing lock file

Package operations: 85 installs, 1 update, 0 removals

  • Installing certifi (2022.12.7)
  • Installing charset-normalizer (3.1.0)
  • Installing frozenlist (1.3.3)
  • Installing idna (3.4)
  • Installing markupsafe (2.1.2)
  • Installing multidict (6.0.4)
  • Installing urllib3 (1.26.15)
  • Installing aiosignal (1.3.1)
  • Installing alabaster (0.7.13)
  • Installing async-timeout (4.0.2)
  • Installing attrs (22.2.0): Installing...
  • Installing babel (2.12.1): Pending...
Connection pool is full, discarding connection: pypi.org. Connection pool size: 10
  • Installing colorama (0.4.6)
  • Installing docutils (0.18.1): Pending...
  • Installing imagesize (1.4.1)
  • Installing jinja2 (3.1.2): Pending...
  • Installing lazy-object-proxy (1.9.0): Pending...
  • Installing packaging (23.0): Pending...
  • Installing pygments (2.14.0): Pending...
  • Installing attrs (22.2.0)
  • Installing babel (2.12.1)
  • Installing colorama (0.4.6)
  • Installing docutils (0.18.1)
  • Installing imagesize (1.4.1)
  • Installing jinja2 (3.1.2)
  • Installing lazy-object-proxy (1.9.0)
  • Installing packaging (23.0)
  • Installing pygments (2.14.0)
  • Installing requests (2.28.2)
  • Updating setuptools (63.2.0 -> 67.6.0)
  • Installing snowballstemmer (2.2.0)
  • Installing sphinxcontrib-applehelp (1.0.4)
  • Installing sphinxcontrib-devhelp (1.0.2)
  • Installing sphinxcontrib-htmlhelp (2.0.1)
  • Installing sphinxcontrib-jsmath (1.0.1)
  • Installing sphinxcontrib-qthelp (1.0.3)
  • Installing typing-extensions (4.5.0)
  • Installing wrapt (1.15.0)
  • Installing sphinxcontrib-serializinghtml (1.1.5)
  • Installing yarl (1.8.2)
  • Installing aiohttp (3.8.4): Installing...
  • Installing astroid (2.15.0): Installing...
  • Installing greenlet (2.0.2): Installing...
  • Installing aiohttp (3.8.4)
  • Installing astroid (2.15.0)
  • Installing greenlet (2.0.2)
  • Installing mypy-extensions (1.0.0)
  • Installing pysocks (1.7.1)
  • Installing python-socks (2.1.1)
  • Installing pyyaml (6.0)
  • Installing six (1.16.0)
  • Installing soupsieve (2.4)
  • Installing sphinx (5.3.0)
  • Installing sphinxcontrib-jquery (2.0.0)
  • Installing tomli (2.0.1)
  • Installing unidecode (1.3.6)
  • Installing aiofiles (22.1.0)
  • Installing aiohttp-socks (0.7.1): Installing...
  • Installing beautifulsoup4 (4.11.2): Installing...
  • Installing dill (0.3.6): Installing...
  • Installing lxml (4.9.2): Pending...
  • Installing mako (1.2.4): Installing...
  • Installing aiohttp-socks (0.7.1)
  • Installing beautifulsoup4 (4.11.2)
  • Installing dill (0.3.6)
  • Installing lxml (4.9.2)
  • Installing mako (1.2.4)
  • Installing mergedeep (1.3.4)
  • Installing mypy (0.991)
  • Installing orjson (3.8.7)
  • Installing python-dateutil (2.8.2)
  • Installing sphinx-autoapi (2.0.1)
  • Installing sphinx-rtd-theme (1.2.0)
  • Installing sqlalchemy (2.0.5.post1)
  • Installing user-agent (0.1.10)
  • Installing zipp (3.15.0)
  • Installing alembic (1.10.2)
  • Installing appdirs (1.4.4)
  • Installing cssselect (1.2.0)
  • Installing importlib-metadata (6.0.0)
  • Installing psycopg2 (2.9.5)
  • Installing pyee (8.2.2)
  • Installing tqdm (4.65.0)
  • Installing ultima-scraper-api (0.0.2.5)
  • Installing websockets (10.4)
  • Installing altgraph (0.17.3)
  • Installing bs4 (0.0.1)
  • Installing click (8.1.3): Installing...
  • Installing fake-useragent (1.1.1): Installing...
Connection pool is full, discarding connection: pypi.org. Connection pool size: 10
  • Installing pathspec (0.11.0): Installing...
  • Installing click (8.1.3)
  • Installing fake-useragent (1.1.1)
  • Installing parse (1.19.0)
  • Installing pathspec (0.11.0)
  • Installing pefile (2023.2.7)
  • Installing platformdirs (3.1.1)
  • Installing pyinstaller-hooks-contrib (2023.0)
  • Installing pyppeteer (1.0.2)
  • Installing pyquery (2.0.0)
  • Installing pywin32-ctypes (0.2.0)
  • Installing ultima-scraper-collection (0.0.0.4): Failed

  FileNotFoundError

  [Errno 2] No such file or directory: 'C:\\Users\\xxxxxx\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\ultima-scraper-ydjPcDtY-py3.10\\Lib\\site-packages\\ultima_scraper_collection\\managers\\database_manager\\connections\\sqlite\\legacy_databases\\messages\\migration\\alembic\\versions\\2c36fcc0b921_content.py'

  at ~\AppData\Local\Programs\Python\Python310\lib\site-packages\poetry\installation\wheel_installer.py:55 in write_to_fs
       51│             # Due to the parallel installation it can happen
       52│             # that two threads try to create the directory.
       53│             os.makedirs(parent_folder, exist_ok=True)
       54│
    →  55│         with open(target_path, "wb") as f:
       56│             hash_, size = copyfileobj_with_hashing(stream, f, self.hash_algorithm)
       57│
       58│         if is_executable:
       59│             make_file_executable(target_path)

  • Installing w3lib (2.1.1)
Script has been updated, exiting in 5 seconds
PS H:\OnlyFans-master>

**Comment by Klutronic** [comment]
I had to run it twice to get it installed as well.  Working great now.

**Comment by wooki0** [comment]
> I had to run it twice to get it installed as well. Working great now.

still not working tried multiple times do i need to download the ultima scraper api?

**Comment by Klutronic** [comment]
I didn't have to.  The only difference I see between you and I is you are running PowerShell.  I did it in an old school command prompt.  It SHOULDN'T make a difference (I am also a PS'er) but who knows...  Does the PS command require poetry like the command line does?  At any rate, this is the command I ran:

C:\Tools\OnlyFans-master>poetry run python .\updater.py
Updating dependencies
Resolving dependencies...

Writing lock file

Package operations: 2 installs, 16 updates, 0 removals

  • Updating charset-normalizer (3.1.0 C:/Users/Drew/AppData/Local/pypoetry/Cache/artifacts/51/f0/38/a244a7e7e295d5c9424ec8b25aaa5b590028344e1aa3ab19a8dba03d64/charset_normalizer-3.1.0-cp311-cp311-win_amd64.whl -> 3.1.0)
  • Updating babel (2.11.0 -> 2.12.1)
  • Updating setuptools (67.3.2 -> 67.6.0)
  • Updating wrapt (1.14.1 -> 1.15.0)
  • Updating astroid (2.14.2 -> 2.15.0)
  • Updating python-socks (2.1.1 -> 2.2.0)
  • Updating orjson (3.8.6 -> 3.8.7)
  • Updating zipp (3.13.0 -> 3.15.0)
  • Updating sqlalchemy (1.4.46 -> 2.0.6)
  • Updating alembic (1.9.4 -> 1.10.2)
  • Installing psycopg2 (2.9.5)
  • Updating importlib-metadata (4.13.0 -> 6.0.0)
  • Updating ultima-scraper-api (0.0.2.3 -> 0.1.3)
  • Updating tqdm (4.64.1 -> 4.65.0)
  • Updating platformdirs (3.0.0 -> 3.1.1)
  • Installing ultima-scraper-collection (0.1.3)
  • Updating pyinstaller (5.8.0 -> 5.9.0)
  • Updating ultima-scraper-renamer (0.0.5 -> 0.1.3)
Script has been updated, exiting in 5 seconds
